app/code/Magento/InventoryCatalogApi/Model/GetSourceCodesBySkusInterface.php

```diff
@@ -0,0 +1,21 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalogApi\Model;
+
+/**
+ * Returns a list of source codes associated to a list of provided skus
+ * @api
+ */
+interface GetSourceCodesBySkusInterface
```

Comment:
I was thinking for a while whether it's okay or not to introduce this interface.

From one side - we can achieve the same result calling `\Magento\InventoryApi\Api\SourceItemRepositoryInterface::getList`
setting up SearchCriteria with a needed set of SKUs

from another side - we can take this service if we consider that this operation would be commonly used As a sugar service.

InventoryCatalogApi is definitely not the right place for this service, as it manipulates only with Sources and Skus (entities from InventoryAPI). Thus, if we decide to introduce one it should be put into `InventoryApi` -> `Inventory` modules.

From another side - we need this service only in `InventoryCatalogAdminUi` module 
and we already have similar service there - `\Magento\InventoryCatalogAdminUi\Model\GetSourceItemsDataBySku`

```
    /**
     * @param string $sku
     *
     * @return array
     */
    public function execute(string $sku): array
    {
        $sourceItemsData = [];

        $searchCriteria = $this->searchCriteriaBuilder
            ->addFilter(SourceItemInterface::SKU, $sku)
            ->create();
        $sourceItems = $this->sourceItemRepository->getList($searchCriteria)->getItems();

        foreach ($sourceItems as $sourceItem) {
            $source = $this->sourceRepository->get($sourceItem->getSourceCode());

            $sourceItemsData[] = [
                SourceItemInterface::SOURCE_CODE => $sourceItem->getSourceCode(),
                SourceItemInterface::QUANTITY => $sourceItem->getQuantity(),
                SourceItemInterface::STATUS => $sourceItem->getStatus(),
                SourceInterface::NAME => $source->getName(),
                'source_status' => $source->isEnabled(),
            ];
        }

        return $sourceItemsData;
    }
```

So, my proposal is to re-use `\Magento\InventoryCatalogAdminUi\Model\GetSourceItemsDataBySku`  and not introducing new service until we have strong reasons. We can discuss this subject, whether we need this service or not.

Reply:
They are similar, but using `GetSourceItemsDataBySku` will require to iterate the whole array, while the new service I added is basically returning with a single query. I did not want to duplicate the code and potentially affect performance.

Reply:
hmm... I did not mention that `SourceRepository` called for any SourceItem in the loop
`$source = $this->sourceRepository->get($sourceItem->getSourceCode());`

I would say that some kind of internal cache should be introduced, something like this

```php
        $sources = [];
        foreach ($sourceItems as $sourceItem) {
            if (!isset($sources[$sourceItem->getSourceCode()])) {
                $sources[$sourceItem->getSourceCode()] = $this->sourceRepository->get($sourceItem->getSourceCode());
            }

            $sourceItemsData[] = [
                SourceItemInterface::SOURCE_CODE => $sourceItem->getSourceCode(),
                SourceItemInterface::QUANTITY => $sourceItem->getQuantity(),
                SourceItemInterface::STATUS => $sourceItem->getStatus(),
                SourceInterface::NAME => $sources[$sourceItem->getSourceCode()]->getName(),
                'source_status' => $sources[$sourceItem->getSourceCode()]->isEnabled(),
            ];
        }
```

as already now we issue unneeded SQL queries building SourceItem grid on Product Page

Reply:
I think we could find a solution in the middle. I agree removing the service, it is not so useful as I was thinking at the beginning. I think we could simply create an internal resource model in `InventoryCatalogAdminUi`. It is probably the fastest and more efficient way.